It is polindrome

hihilisk opened the pull request.

hihilisk committed:
refactoring code

hihilisk committed:
(added folder`s to ingore entries)

hihilisk committed:
refactoring code v2

hihilisk committed:
created function palindrome on JS

hihilisk committed:
refactoring code

hihilisk commented on palindrome/palindrome.rb:
```diff
@@ -1,15 +1,8 @@
+    'It`s a palindrome, no spaces'
   else
     'It`s not a palindrome'
   end
```
в ходе написания функции я записывал значения в переменную и возвращал ее через return (дабы убедиться что все работает штатно) , так и собрал ее, а потом до меня дошло что у меня в принципе нет необходимости хранить результаты преобразований

hihilisk commented on palindrome/palindrome.js:
```diff
@@ -1,11 +1,7 @@
+  } else if (string.replace(/\s/g,'') === string.replace(/\s/g,'').split("").reverse().join("") ) {
       return 'It`s a palindrome, no spaces';
   } else {
       return 'It`s not a palindrome';
```
const palindrome = string => (string === string.split("").reverse().join("")) ? 'It`s a palindrome' : (string.replace(/\s/g,'') === string.replace(/\s/g,'').split("").reverse().join("")) ? 'It`s a palindrome, no spaces' : 'It`s not a palindrome';

и оно работает))))

hihilisk replied:
в ходе написания функции я записывал значения в константу и возвращал ее через return (дабы убедиться что все работает штатно) , так и собрал ее, а потом до меня дошло что у меня в принципе нет необходимости хранить результаты преобразований

alexander-maxa commented on palindrome/palindrome.js:
```diff
@@ -0,0 +1,9 @@
+function palindrome (string) {
```
function palindrome(value) {

alexander-maxa commented on palindrome/palindrome.js:
```diff
@@ -0,0 +1,9 @@
+function palindrome (string) {
+  if (string === string.split("").reverse().join("")) {
```
string === string.split('').reverse().join('')

alexander-maxa commented on palindrome/palindrome.js:
```diff
@@ -0,0 +1,9 @@
+function palindrome (string) {
+  if (string === string.split("").reverse().join("")) {
+      return 'It`s a palindrome';
+  } else if (string.replace(/\s/g,'') === string.replace(/\s/g,'').split("").reverse().join("") ) {
```
string.replace(/\s/g,'' - вынеси в переменную

alexander-maxa commented on palindrome/palindrome.rb:
```diff
@@ -1,15 +1,8 @@
-  string_no_spaces = string.gsub(' ', '')
-  if string_no_spaces == string_no_space.reverse
-    'It`s a palindrome'
+  elsif string.gsub(' ', '') == string.reverse.gsub(' ', '')
```
Может быть сразу удалить пробелы, чтобы потом не делать это два раза?

hihilisk committed:
refactoring code

hihilisk committed:
refactoring code palindrome.rb

alexander-maxa commented on palindrome/palindrome.js:
```diff
@@ -0,0 +1,10 @@
+function palindrome(string) {
+  let stringReplaceSpace=string.replace(/\s/g,'');
+  if (string===string.split('').reverse().join('')) {
```
пробелы?

alexander-maxa commented on palindrome/palindrome.js:
```diff
@@ -0,0 +1,10 @@
+function palindrome(string) {
+  let stringReplaceSpace=string.replace(/\s/g,'');
```
строкаЗаменитьПробелы - это больше похоже на метод

Во-первых, не string, а value (string очень созвучно с типом) я об этом писал, во вторых старайся не использовать глаголы в именах переменных. Можно было бы назвать newValue or valueWithNoSpaces

hihilisk committed:
refactoring code palindrome.js
renamed the variable 'string' to 'value'
returned spaces to operands
removed unnecessary 'else'

hihilisk committed:
create sorting function

hihilisk committed:
Revert "create sorting function"

This reverts commit 394425f7

hihilisk committed:
refactoring code

alexander-maxa commented on palindrome/palindrome.js:
```diff
@@ -0,0 +1,13 @@
+function palindrome(value) {
+  let valueWithNoSpaces = value.replace(/\s/g,'');
```
const

alexander-maxa replied:
value.replace(/\s/g, '');

alexander-maxa commented on palindrome/palindrome.rb:
```diff
@@ -1,15 +1,9 @@
-  if string_no_spaces == string_no_space.reverse
+  if string == string.reverse
     'It`s a palindrome'
+  elsif string_no_spaces == string_no_spaces.reverse
```
string_with_no_spaces

hihilisk committed:
refactoring code inside: 'palindrome.js' and 'palindrome.rb'
